nautobot/extras/api/serializers.py

```diff
@@ -1032,11 +1048,21 @@ class Meta:
             "color",
             "description",
             "tagged_items",
+            "content_types",
             "custom_fields",
             "created",
             "last_updated",
         ]
 
+    def validate(self, attr):
+        data = super().validate(attr)
+
+        # All relevant content_types should be assigned to tag without content_types
```

Comment:
🤷 I don't know if there is a clear answer for this...

I think an "All" choice would be nice to see in a choice list and be a magic word that during validate triggers the below.

Then again, I also don't see any case where a Tag with no assigned content types is valuable so all content types might be an acceptable default for none.

Then again again, what will be the expected behavior for users when new models are added (either via a plugin or in newer versions) when they haven't specifically assigned tags to content types.

Reply:
Yeah I'm still on the fence about this one myself. I do think it violates least-surprise principles for "none" to mean "all". You're right that a tag with *no* content-types is not useful, which is why I'd personally prefer for the ContentTypeField to have `required=True` so as to make that an explicit part of the REST API and UI contracts.

In the UI it's relatively simple to select all options (click first item in the list, scroll to bottom of list, shift-click last item in the list) if that's what's really desired but my *belief* is that in most cases a tag does *not* logically apply to every possible model type and a user intentionally selecting only the applicable types is a better practice to guide users towards in the vast majority of cases.

I'm not sure offhand if we have an obvious REST API endpoint for "get me the list of all possible tag content-types" (though I think we should have that info available in the Swagger/OpenAPI docs) so we might need to think about how best to make it possible for the REST API client to easily identify and select "all content-types" when creating/updating a tag.

Not having a magic "all" option also makes the new-model behavior more explicit - when a new model is added, it's going to be on the user to enable whatever tag(s) make sense for that model, we won't do any magic fixup. (At least in the general case - I can certainly see cases where splitting/merging model classes would make a certain degree of fixup as part of the migration make sense, for example when we implement #1178, any tags on VirtualMachine should be automatically fixed up to become tags on Device, etc.)